repository/src/main/java/org/apache/atlas/ESAliasStore.java

```diff
@@ -0,0 +1,270 @@
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.getAssets;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.getConnectionId;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.getIsAllow;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.getQualifiedName;
+import static org.apache.atlas.accesscontrol.persona.AtlasPersonaUtil.mapOf;
+import static org.apache.atlas.accesscontrol.purpose.AtlasPurposeUtil.getTags;
+import static org.apache.atlas.repository.Constants.PROPAGATED_TRAIT_NAMES_PROPERTY_KEY;
+import static org.apache.atlas.repository.Constants.QUALIFIED_NAME;
+import static org.apache.atlas.repository.Constants.TRAIT_NAMES_PROPERTY_KEY;
+import static org.apache.atlas.repository.Constants.VERTEX_INDEX_NAME;
+
+@Component
+public class ESAliasStore implements IndexAliasStore {
+    private static final Logger LOG = LoggerFactory.getLogger(ESAliasStore.class);
+
+    private final AtlasGraph graph;
+    private final EntityGraphRetriever entityRetriever;
+
+    @Inject
+    public ESAliasStore(AtlasGraph graph,
+                        EntityGraphRetriever entityRetriever) {
+        this.graph = graph;
+        this.entityRetriever = entityRetriever;
+    }
+
+    @Override
+    public boolean createAlias(PersonaContext personaContext) throws AtlasBaseException {
+        String aliasName = getAliasName(personaContext.getPersona());
+
+        ESAliasRequestBuilder requestBuilder = new ESAliasRequestBuilder();
+        requestBuilder.addAction(ADD, new AliasAction(VERTEX_INDEX_NAME, aliasName));
+
+        graph.createOrUpdateESAlias(requestBuilder);
+        return true;
+    }
+
+    @Override
+    public boolean updateAlias(PersonaContext personaContext) throws AtlasBaseException {
+        String aliasName = getAliasName(personaContext.getPersona());
+
+        Map<String, Object> filter = getFilter(personaContext);
+
+        ESAliasRequestBuilder requestBuilder = new ESAliasRequestBuilder();
+        requestBuilder.addAction(ADD, new AliasAction(VERTEX_INDEX_NAME, aliasName, filter));
+
+        graph.createOrUpdateESAlias(requestBuilder);
+
+        return true;
+    }
+
+    @Override
+    public boolean createAlias(PurposeContext purposeContext) throws AtlasBaseException {
+        String aliasName = getAliasName(purposeContext.getPurpose());
+
+        ESAliasRequestBuilder requestBuilder = new ESAliasRequestBuilder();
+        requestBuilder.addAction(ADD, new AliasAction(VERTEX_INDEX_NAME, aliasName));
+
+        graph.createOrUpdateESAlias(requestBuilder);
+        return true;
+    }
+
+    @Override
+    public boolean updateAlias(PurposeContext purposeContext) throws AtlasBaseException {
+        String aliasName = getAliasName(purposeContext.getPurpose());
+
+        Map<String, Object> filter = getFilter(purposeContext);
+
+        ESAliasRequestBuilder requestBuilder = new ESAliasRequestBuilder();
```

Comment:
can be initialized in the constructor since the same initialization is executed in each operation.

Reply:
requestBuilder maintains state of the request
if we initialize it in constructor, since ESAliasStore is injected as singleton throughout the application, requestBuilder's state will be shared throughout the application

To get fresh state of requestBuilder, initializing it whenever needed